lib/Extension/LanguageServerWorseReflection/DiagnosticProvider/WorseDiagnosticProvider.php

```diff
@@ -94,4 +98,26 @@ private static function toLspTags(array $tags): array
             $tags
         );
     }
+
+    private function diagnosticToCode(Diagnostic $diagnostic): string
```

Comment:
instead of guessing the name we could set it explicitly. the reason I didn't was because of the existing `DiagnosticProvider#name` method - this method however could probably simply be removed.

Reply:
Does this also relate to this: #2074 ?

Reply:
Nope